domain/application/service/application.go

```diff
@@ -344,3 +355,34 @@ func (s *ApplicationService) validateStorageDirectives(ctx context.Context, mode
 	}
 	return nil
 }
+
+// WatchableApplicationService provides the API for working with applications and the
+// ability to create watchers.
+type WatchableApplicationService struct {
+	ApplicationService
+	watcherFactory WatcherFactory
+}
+
+// NewWatchableApplicationService returns a new service reference wrapping the input state.
+func NewWatchableApplicationService(st ApplicationState, watcherFactory WatcherFactory, registry storage.ProviderRegistry, logger logger.Logger) *WatchableApplicationService {
+	return &WatchableApplicationService{
+		ApplicationService: ApplicationService{
+			st:       st,
+			registry: registry,
+			logger:   logger,
+		},
+		watcherFactory: watcherFactory,
+	}
+}
+
+// WatchApplicationUnitLife returns a watcher that observes changes to the life of any units if an application.
+func (s *WatchableApplicationService) WatchApplicationUnitLife(appName string) (watcher.StringsWatcher, error) {
+	lifeGetter := func(ctx context.Context, db coredatabase.TxnRunner, ids ...string) (map[string]life.Life, error) {
+		l, err := s.st.GetApplicationUnitLife(ctx, appName, ids...)
+		return l, err
+	}
+	lifeMapper := domain.LifeStringsWatcherMapperFunc(s.logger, lifeGetter)
+
+	table, query := s.st.InitialWatchStatementUnitLife(appName)
+	return s.watcherFactory.NewNamespaceMapperWatcher(table, changestream.All, query, lifeMapper)
```

Comment:
That's no what @ycliuhw is asking, can we get away with `changestream.Delete`

Reply:
Why? The watcher needs to know about all changes so it can emit an event when the life goes from alive to dying for example.

Reply:
Correct, that was my mistake. I think it's because we're only calling out that it's doing something with Delete in the mapper, then that's its only concern. That is just _one_ of its concerns.